pkg/objstore/s3/s3.go

```diff
@@ -51,7 +52,14 @@ func (conf *Config) Validate() error {
 
 // NewBucket returns a new Bucket using the provided s3 config values.
 func NewBucket(conf *Config, reg prometheus.Registerer) (*Bucket, error) {
-	client, err := minio.NewCore(conf.Endpoint, conf.AccessKey, conf.SecretKey, !conf.Insecure)
+	var f func(string, string, string, bool) (*minio.Client, error)
+	if conf.SignatureV2 {
+		f = minio.NewV2
```

Comment:
I would rather go for:
```
if v2:
   client, err := minio.NewV2(.. .. .. ..)
else 
   client, err := minio.NewV4(.. .. .. ..)

// check err
```
But it might be my personal opinion - not a blocker

Reply:
I considered that, but wasn't keen because you end up duplicating the function arguments, which include the negation on `conf.Insecure`.

I agree that calling `f` is a bit funky, but I think that it's cleaner. Happy to go with your preference.

Reply:
I think it's fine, only 4, clear args (even with this insecure negation), but it's up to you (:

Reply:
I'd rather leave it as-is if you have no strong preference.